src/main/scala/molmed/apps/SolidGenerateXMLfromCSV.scala

```diff
@@ -0,0 +1,80 @@
+package molmed.apps
+
+import java.io.File
+import scala.io.Source
+import molmed.utils.GeneralUtils
+import java.io.PrintWriter
+import java.nio.file.Files
+import java.nio.file.Paths
+
+/**
+ * Utility program to convert generate AfMD meta data in a tsv file to UU SNP format.
+ */
+object AmfdGenerateXMLfromCSV extends App {
+
+  val usage = "Missing input file. Usage is: java -classpath <classpath> molmed.apps.MakeAfMDInputDataHierarchy <meta file> <new root dir>"
+
+  val metaFile = try {
+    new File(args(0))
+  } catch {
+    case e: ArrayIndexOutOfBoundsException =>
+      throw new IllegalArgumentException(usage)
+  }
+
+  val rootDir = try {
+    new File(args(1))
+  } catch {
+    case e: ArrayIndexOutOfBoundsException =>
+      throw new IllegalArgumentException(usage)
+  }
+
+  rootDir.mkdirs()
+  val reportFile = new File(rootDir + "/report.tsv")
+  val reportWriter = new PrintWriter(reportFile)
+
+  reportWriter.println(List("#SampleName", "Lane", "ReadLibrary", "FlowcellId").mkString("\t"))
+
+  private val lines = Source.fromFile(metaFile).getLines.map(line => {
+    val elements = line.split("\\s+")
+    elements
+  })
+
+  for (line <- lines) {
+    val sampleName = line(0)
+    val flowCellId = line(1)
+    val lane = line(2).toInt
+    val index = line(3)
+
+    // take all the last columns as input sequence files.
+    val seqFiles = line.drop(4)
+
+
+    // Sometime the meta data file contains more files than the folder...
```

Comment:
Good idea. However, currently I do not use this class so I will improve when there is an obvious need to make it work.